src/lxc/lxccontainer.c

```diff
@@ -4659,6 +4680,158 @@ static bool do_lxcapi_restore(struct lxc_container *c, char *directory, bool ver
 
 WRAP_API_2(bool, lxcapi_restore, char *, bool)
 
+static int do_lxcapi_mount(struct lxc_container *c,
+			const char *source, const char *target,
+			const char *filesystemtype, unsigned long mountflags,
+			const void *data, struct lxc_mount *mnt) {
+	int fd;
+	char *template, *suff, *path;
+	pid_t pid, init_pid;
+	size_t len;
+	struct stat sb;
+	int ret = -1;
+
+	if (!c || !c->lxc_conf) {
+		ERROR("Container or configuration is NULL");
+		goto out_err;
+	}
+
+	len = strlen(c->lxc_conf->lxc_shmount.path_host) + sizeof("/lxcmountXXXXXX");
+	template = malloc(len);
+	if (!template) {
+		SYSERROR("Not enough memory");
+		goto out_err_free;
+	}
+	ret = snprintf(template, len, "%s/lxcmountXXXXXX", c->lxc_conf->lxc_shmount.path_host);
+	if (ret < 0 || (size_t)ret >= len) {
+		SYSERROR("Error writing shmounts tempdir name");
+		goto out_err_free;
+	}
+
+	/* Create a temporary dir / file
+	 * under the shared mountpoint
+	 */
+	ret = stat(source, &sb);
+	if (ret < 0) {
+		SYSERROR("Could not read source information");
+		goto out_err_free;
+	}
+
+	if (S_ISREG(sb.st_mode)) {
+		fd = mkstemp(template);
```

Comment:
Seems you're leaking the `fd`. This needs to be closed when `fd >= 0`.
So at the top of the function do:

    int fd = -EBADF;

Then when you jump to `out_err_free` add a condition:

    if (fd >= 0)
            close(fd);